plugins/aws/sts_provisioner.go

```diff
@@ -231,11 +243,7 @@ func resolveLocalAnd1PasswordConfigurations(itemFields map[sdk.FieldName]string,
 	}
 
 	if awsConfig.HasMfaSerial() && awsConfig.MfaToken == "" {
-		return fmt.Errorf("MFA failed: an MFA serial was found but no OTP has been set up in 1Password")
-	}
-
-	if !awsConfig.HasMfaSerial() && awsConfig.MfaToken != "" {
-		return fmt.Errorf("MFA failed: an OTP was found wihtout a corresponding MFA serial")
+		return fmt.Errorf("MFA failed: the selected profile requires an OTP because an MFA serial (%s) was detected but no 'One-Time Password' field was found in the associated item", awsConfig.MfaSerial)
```

Comment:
@Lucy (couldn't find your official handle) do error messages like this sound fine? The question applies for all error messages in this function, which users could be seeing while setting things up.

Reply:
Am I understanding correctly that this error would be returned when an MFA serial is found on the 1P item, but no one-time password field is found?

If so, I wonder if it would be more helpful to explain the actions to take to fix this (adding an OTP or removing the serial) than to explain that the existence of the MFA serial prompted this.

I'm not sure if we typically include docs links in error messages, so here are two options with or without a learn more link. Please feel free to edit these as needed! 

MFA failed: an MFA serial (%s) was detected on the associated item for the selected profile, but no 'One-Time Password' field was found. Learn how to add an OTP field to your item: https://developer.1password.com/docs/cli/shell-plugins/aws/#optional-set-up-multi-factor-authentication Or remove the MFA serial field to proceed without multi-factor authentication.

MFA failed: an MFA serial (%s) was detected on the associated item for the selected profile, but no 'One-Time Password' field was found. Add an OTP to your item to use multi-factor authentication, or remove the MFA serial field to proceed without MFA.

Reply:
Done, thanks!